pinot-core/src/main/java/org/apache/pinot/core/segment/processing/partitioner/TableConfigPartitioner.java

```diff
@@ -41,6 +43,16 @@ public TableConfigPartitioner(String columnName, ColumnPartitionConfig columnPar
 
   @Override
   public String getPartition(GenericRow genericRow) {
-    return String.valueOf(_partitionFunction.getPartition(genericRow.getValue(_column)));
+    return String.valueOf(_partitionFunction.getPartition(convertToString(genericRow.getValue(_column))));
+  }
+
+  private static String convertToString(Object value) {
```

Comment:
Is there a way to catch unsupported types here to ensure that for new types, we do the string conversion correctly ?  Same for the logic in **MutableSegmentImpl.java** (similar to what you have in getStringValue utility)

Reply:
We can potentially check for all possible classes supported (e.g. Integer, Long, Float, Double, String etc.), but that might involve performance overhead (one if check per type). Since we don't add new data types frequently, I prefer keeping the check only for types that need to be converted

Reply:
Ok sounds good. I assume that if a new type is introduced, we anyway have to modify RequestContextUtils to add the new literal type? (where the exception is thrown). if thats the case, we can consider moving this logic and the one in MutableSegmentImpl.java to RequestContextUtils, so that they are in the same file and easy to discover all the places that needs the change.

Reply:
Good point. Moved them into `FieldSpec` where the `DataType` is defined